protocol-units/da/m1/light-node/src/v1/sequencer.rs

```diff
@@ -54,23 +54,70 @@ impl LightNodeV1Operations for LightNodeV1 {
 
 impl LightNodeV1 {
 	pub async fn tick_block_proposer(&self) -> Result<(), anyhow::Error> {
-		let block = self.memseq.wait_for_next_block().await?;
-		match block {
-			Some(block) => {
-				info!("Built block {:?} with {:?} transactions", block.id(), block.transactions.len());
-				let block_blob = self.pass_through.create_new_celestia_blob(
-					serde_json::to_vec(&block)
-						.map_err(|e| anyhow::anyhow!("Failed to serialize block: {}", e))?,
-				)?;
-
-				let height = self.pass_through.submit_celestia_blob(block_blob).await?;
-
-				debug!("Submitted block: {:?} {:?}", block.id(), height);
-			}
-			None => {
-				// no transactions to include
+
+		let half_building_time = self.memseq.building_time_ms();
+		let start = std::time::Instant::now();
+
+		let memseq = self.memseq.clone();
+		// should help performance by dedicating a thread to this
+		let blocks = tokio::spawn(async move {
+			let mut blocks = Vec::new();
+			while (start.elapsed().as_millis() as u64)  < half_building_time {
+
+				// this has an internal timeout based on its building time
+				// so in the worst case scenario we will roughly double the internal timeout
+				let uid = uuid::Uuid::new_v4();
```

Comment:
See my comments above to @mzabaluev. I needed an ID prior to have a `block_id` for the logging. There may be a better way as tracing spans generate their own UID.

Reply:
Yes, each span already has an internal ID. If you manage to pass the span throughout your flow, you should see it in all events that are descendants in the span (assuming it is enabled), and also in the synthetic close event that I specified the timing subscriber to log. But I don't know if many subscribers display the span ID; also, assigning span IDs is the responsibility of the subscriber and it may in principle not be unique.

I'd favor using explicitly generated IDs if you must have something to identify your flows. But it could be as simple as an (atomic) `u64` counter, no need to invoke a whole crypto-grade RNG for this.

Reply:
I agree with @mzabaluev. A counter can be enough.

Reply:
Okay, counter with wrapping add coming up. 

I'll leave the spanning for a future PR.